db/db_path_supplier.h

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+
+#include "db/column_family.h"
+#include "rocksdb/status.h"
+
+#define UNUSED(expr) do { (void)(expr); } while (0)
+
+namespace rocksdb {
+
+// An object vendored by column family to dynamically supply db path to
+// functions that need to decide which db_path to flush an sst file to. The
+// supplier object is mutable (because you can add file size to it) and
+// can update global file counters, so usage should be inside proper locking.
+class DbPathSupplier {
+ public:
+  explicit DbPathSupplier(const ImmutableCFOptions& ioptions):
+    ioptions_(ioptions) {}
+
+  virtual ~DbPathSupplier() = default;
+
+  Status FsyncDbPath(uint32_t path_id) const;
+
+  virtual uint32_t GetPathId(int level) const {
+    UNUSED(level);
+    return 0;
+  }
+
+  // Is the given path_id an acceptable path_id
+  // for this supplier?
+  //
+  // This method is used in compaction to decide
+  // if it is feasible to only change the level
+  // of an sst file without actually moving the
+  // data (if it's a trivial move).
+  //
+  // For a random path supplier, for example,
+  // it doesn't matter which path_id is given
+  // because path_ids are chosen randomly anyway.
+  // For a fix path supplier, however, the given
+  // path_id really needs to match the fixed
+  // path_id in order for us to say it's trivial.
+  virtual bool AcceptPathId(
+      uint32_t path_id, int output_level) const {
+    UNUSED(path_id);
+    UNUSED(output_level);
+    return false;
+  }
+
+ protected:
+  const ImmutableCFOptions ioptions_;
```

Comment:
Same here. `uint32_t /* path_id */, int /*output_level*/`.

In fact, do we need to define GetPathId() and AcceptPathId() here at all? Can we instead make them ` = 0`?

Reply:
Fixed. Also took your advice and make them `=0`.